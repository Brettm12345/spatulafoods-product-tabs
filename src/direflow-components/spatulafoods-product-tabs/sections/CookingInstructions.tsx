import {FC} from 'react';
import {Product} from '../data/products';

export const CookingInstructions: FC<Product> = ({cookingInstructions}) => (
  <div className="prose prose-lg md:prose-xl mx-auto px-4 md:px-0">
    {cookingInstructions}
  </div>
);
