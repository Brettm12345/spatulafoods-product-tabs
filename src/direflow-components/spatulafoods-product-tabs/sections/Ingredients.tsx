import React, {FC} from 'react';
import {Product} from '../data/products';

export const Ingredients: FC<Product> = ({ingredients}) => (
  <ul className="col-gap-8 column-count-3 min-w-full list-none space-y-2">
    {ingredients.map(ingredient => (
      <li key={ingredient} className="text-sm md:text-base">
        {ingredient}
      </li>
    ))}
  </ul>
);
