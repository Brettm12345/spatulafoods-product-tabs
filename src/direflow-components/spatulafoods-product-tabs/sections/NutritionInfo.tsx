import React, {FC, Fragment} from 'react';
import classNames from 'classnames';
import {isCategory, Product} from '../data/products';

const rowPadding = 'p-2 md:p-4';
const numericRow = classNames('text-end', rowPadding);
const row = classNames('text-start', rowPadding);

interface NutritionRowProps {
  dailyValue?: number;
  content: string;
  ingredient: string;
  isSubIngredient?: boolean;
  isLastSubIngredient?: boolean;
}

const NutritionRow: FC<NutritionRowProps> = ({
  dailyValue,
  content,
  ingredient,
  isSubIngredient = false,
  isLastSubIngredient = false,
}) => (
  <tr
    className={classNames(
      {'border-b': !isSubIngredient || isLastSubIngredient},
      'border-gray-200'
    )}
  >
    <td
      className={classNames(
        row,
        isSubIngredient ? 'pl-8 md:pl-8' : 'font-bold'
      )}
    >
      {ingredient}
    </td>
    <td className={numericRow}>{content}</td>
    <td className={numericRow}>{dailyValue ? `${dailyValue}%` : ''}</td>
  </tr>
);

export const NutritionInfo: FC<Product> = ({nutrition}) => (
  <table className="table-auto w-full text-sm md:text-base">
    <tr className="tracking-wider bg-gray-200 uppercase">
      <th className={row}>Ingredient</th>
      <th className={numericRow}>Amount</th>
      <th className={numericRow}>% DV</th>
    </tr>
    {Object.entries(nutrition).map(([name, value]) => {
      if (isCategory(value)) {
        const {breakdown, total} = value;
        return (
          <Fragment key={name}>
            <NutritionRow
              dailyValue={total.dailyValue}
              content={total.content}
              ingredient={name}
            />
            {Object.entries(breakdown).map(([name, value], index, arr) => (
              <NutritionRow
                key={name}
                dailyValue={value.dailyValue}
                content={value.content}
                ingredient={name}
                isSubIngredient
                isLastSubIngredient={index === arr.length - 1}
              />
            ))}
          </Fragment>
        );
      }
      const {dailyValue, content} = value;
      return (
        <NutritionRow
          key={name}
          dailyValue={dailyValue}
          content={content}
          ingredient={name}
        />
      );
    })}
  </table>
);
