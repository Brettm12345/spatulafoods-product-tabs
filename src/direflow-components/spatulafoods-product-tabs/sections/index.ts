import type {FC} from 'react';
import type {Product} from '../data/products';
import {CookingInstructions} from './CookingInstructions';
import {NutritionInfo} from './NutritionInfo';
import {Ingredients} from './Ingredients';

type Sections = Record<string, FC<Product>>;
export const sections: Sections = {
  'Cooking Instructions': CookingInstructions,
  'Nutrition Info': NutritionInfo,
  Ingredients: Ingredients,
};
