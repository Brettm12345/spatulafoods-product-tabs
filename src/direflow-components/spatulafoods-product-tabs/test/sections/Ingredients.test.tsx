import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import {Ingredients} from '../../sections/Ingredients'
import { product } from './__mocks__/product';

it('renders without crashing', () => {
  const div = document.createElement('div');
  render(<Ingredients {...product} />, div)
  unmountComponentAtNode(div)
})

it('matches snapshot as expected', () => {
  const renderTree = renderer.create(<Ingredients {...product} />)
  expect(renderTree).toMatchSnapshot()
})