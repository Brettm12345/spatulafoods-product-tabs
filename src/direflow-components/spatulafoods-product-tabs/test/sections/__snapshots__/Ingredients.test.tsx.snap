// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot as expected 1`] = `
<div
  className="flex flex-col mx-auto px-1 text-lg leading-relaxed space-y-4 md:px-0"
>
  <div>
    <b>
      Ingredients
      :
    </b>
     
    Chicken broth, Green beans, Spanish rice, Shrimp peeled tail-on, Chicken thigh boneless skinless, Onion yellow, Chorizo, Mussels precooked, Calamari rings, Tomatoes, Wine white cooking (Wine, Salt, Sulphites), Olive oil, Red bell pepper, Yellow bell pepper, Garlic fresh, Smoked paprika, Canola Oil, Parsley fresh, Kosher salt, Black pepper ground, Bay Leaf, Saffron
  </div>
  <div>
    <b>
      Contains
      :
    </b>
     
    Milk, Soy beans, Wheat, Molluscs, Sesame seeds, Sulphites
  </div>
</div>
`;
